scene/animation/animation_node_state_machine.cpp

```diff
@@ -1816,6 +1905,15 @@ void AnimationNodeStateMachine::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_node_position", "name", "position"), &AnimationNodeStateMachine::set_node_position);
 	ClassDB::bind_method(D_METHOD("get_node_position", "name"), &AnimationNodeStateMachine::get_node_position);
 
+	ClassDB::bind_method(D_METHOD("set_node_visibility", "name", "node_visibility"), &AnimationNodeStateMachine::set_node_visibility);
+	ClassDB::bind_method(D_METHOD("get_node_visibility", "name"), &AnimationNodeStateMachine::get_node_visibility);
+
+	ClassDB::bind_method(D_METHOD("set_node_group_selections", "name", "selection index", "node_group_selections"), &AnimationNodeStateMachine::set_node_group_selections);
+	ClassDB::bind_method(D_METHOD("get_node_group_selections", "name"), &AnimationNodeStateMachine::get_node_group_selections);
+
+	ClassDB::bind_method(D_METHOD("set_node_group_names", "node_group_names"), &AnimationNodeStateMachine::set_node_group_names);
+	ClassDB::bind_method(D_METHOD("get_node_group_names"), &AnimationNodeStateMachine::get_node_group_names);
```

Comment:
By the way, these do not need to be TOOLS_ENABLED. These and the registration to properties are necessary for the consistency of data serialized as .tres.

Reply:
I will be honest I can't follow all of these I will commit all of them and see how it goes.